ksqldb-engine/src/main/java/io/confluent/ksql/util/BinPackedPersistentQueryMetadataImpl.java

```diff
@@ -290,10 +295,9 @@ public String getQueryApplicationId() {
     return applicationId;
   }
 
-  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "topology is for reference")
   @Override
   public NamedTopology getTopology() {
-    return topology;
+    return namedTopologySupplier.get();
```

Comment:
Was  `getTopology` ever being called before this? If so then it probably wants the actual NamedTopology for that query, which is important. How about adding a separate method for this, eg `getTopologyCopy`

Reply:
I don't think that it really needs the original. But lets add a separate method just incase